aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -29,9 +29,32 @@
 struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct aesd_circular_buffer *buffer,
             size_t char_offset, size_t *entry_offset_byte_rtn )
 {
-    /**
-    * TODO: implement per description
-    */
+    size_t acc, count;
+    size_t offset_curr = buffer->out_offs;
+    acc = 0;
+    count = 0;
+
+    if (buffer == NULL) return NULL;
+
+    while(count < AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED)
+    {
+        if(char_offset < acc + buffer->entry[offset_curr].size) 
+        {
+            *entry_offset_byte_rtn = char_offset-acc;
+            return &(buffer->entry[offset_curr]);
+        }
+
+        acc += buffer->entry[offset_curr].size;
+        offset_curr = (offset_curr + 1) % AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
+        count++;
+
+        if(offset_curr == buffer->in_offs && !buffer->full){
```

Comment:
I don't think the `buffer->full` check is relevant here... anytime we increment past the in_offs we'll be exceeding the buffer length.